packages/components/src/RichTextEditor/RichTextEditor/subcomponents/ToolbarControls/ToolbarControls.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback } from 'react'
+import { useIntl } from '@cultureamp/i18n-react-intl'
+import type { ToolbarItems } from '../../../types'
+import type { CommandOrTransaction } from '../../../utils/core'
+import type { ProseMirrorModel, ProseMirrorState } from '../../../utils/prosemirror'
+import { useControlMap } from '../../utils/controlmap'
+import { ToggleIconButton } from '../ToggleIconButton'
+import { Toolbar } from '../Toolbar'
+import { ToolbarSection } from '../ToolbarSection'
+
+type ToolbarControlsProps = {
+  editorId: string
+  controls?: ToolbarItems[]
+  editorState: ProseMirrorState.EditorState
+  schema: ProseMirrorModel.Schema<any, any>
+  dispatchTransaction: (commandOrTransaction: CommandOrTransaction) => void
+}
+
+export const ToolbarControls = ({
+  editorId,
+  controls,
+  editorState,
+  schema,
+  dispatchTransaction,
+}: ToolbarControlsProps): JSX.Element | null => {
+  const controlMap = useControlMap(schema, editorState, controls)
+  const { formatMessage } = useIntl()
+  const handleControlClick = useCallback(
+    (action: ProseMirrorState.Command) => {
+      dispatchTransaction(action)
+    },
+    [dispatchTransaction],
+  )
+
+  if (controlMap.length <= 0) return null
+
+  return (
+    <Toolbar
```

Comment:
I noticed a visual diff in the error state snapshot, which likely comes from the className additional error styles pass into the `Toolbar` `classNameOverride` not being there (See screenshot below).

<img width="1725" height="250" alt="Screenshot 2025-10-28 at 9 30 38 am" src="https://github.com/user-attachments/assets/7a71c028-f007-49b6-ace8-d99568560c3b" />

<img width="597" height="155" alt="Screenshot 2025-10-28 at 9 32 57 am" src="https://github.com/user-attachments/assets/5756f8ba-feae-49e5-a267-6b83aaa14fa8" />